src/main/kotlin/org/rust/cargo/project/module/persistence/impl/CargoModuleTargetsServiceImpl.kt

```diff
@@ -0,0 +1,54 @@
+package org.rust.cargo.project.module.persistence.impl
+
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.components.Storage
+import com.intellij.openapi.components.StoragePathMacros
+import org.rust.cargo.CargoProjectDescription
+import org.rust.cargo.project.module.persistence.CargoModuleTargetsService
+import java.util.*
+
+@State(
+    name = "CargoModuleTargets",
+    storages = arrayOf(Storage(file = StoragePathMacros.MODULE_FILE))
+)
+class CargoModuleTargetsServiceImpl : PersistentStateComponent<CargoModuleTargetsServiceImpl.State>
+                                    , CargoModuleTargetsService {
+
+    private var myState: State = State()
+
+    // IDEA serializer requires State components to have default constructors,
+    // nullable fields and mutable collections. `null`s in fields signal
+    // serialization failure, so we can't `!!` here safely.
+    data class State(
+        var targets: MutableCollection<Target> = ArrayList()
+    )
+
+    data class Target(
+        var path: String? = null,
+        var kind: CargoProjectDescription.TargetKind? = null
+    ) {
+
+        fun intoTarget(): CargoProjectDescription.Target? {
```

Comment:
No, I can't make this private, or serializer will fail at run time. But I still brushed up this a little.